qa/rpc-tests/wallet.py

```diff
@@ -346,5 +348,60 @@ def run_test (self):
+
+        # Get all non-zero utxos together
+        chain_addrs = [self.nodes[0].getnewaddress(), self.nodes[0].getnewaddress()]
+        singletxid = self.nodes[0].sendtoaddress(chain_addrs[0], self.nodes[0].getbalance(), "", "", True)
+        self.nodes[0].generate(1)
+        node0_balance = self.nodes[0].getbalance()
+        # Now split into two chains
+        rawtx = self.nodes[0].createrawtransaction([{"txid":singletxid, "vout":0}], {chain_addrs[0]:self.nodes[0].getbalance()/2-Decimal('0.01'), chain_addrs[1]:self.nodes[0].getbalance()/2-Decimal('0.01')})
+        signedtx = self.nodes[0].signrawtransaction(rawtx)
+        singletxid = self.nodes[0].sendrawtransaction(signedtx["hex"])
+        txids = [singletxid, singletxid]
+        # Swap output addrs if not aligned
+        if chain_addrs[0] == self.nodes[0].getrawtransaction(singletxid, True)["vout"][1]["scriptPubKey"]["addresses"][0]:
+            chain_addrs = [chain_addrs[1], chain_addrs[0]]
+
+        # Use sendtoaddress with only one output at a time, successfully
+        vout = 1
+        for chainindex in range(chainlimit-1):
+            for i in range(2):
+                self.nodes[0].lockunspent(False, [{"txid":txids[(i+1)%2], "vout":vout}])
+                amount = self.nodes[0].listunspent(0, 0, [chain_addrs[i]])[0]["amount"]
+                txids[i] = self.nodes[0].sendtoaddress(chain_addrs[i], amount, "", "", True)
+                self.nodes[0].lockunspent(True, [{"txid":txids[(i+1)%2], "vout":vout}])
+
+                vout = 0
+                node0_balance = self.nodes[0].getbalance()
+
+                # If combining should fail, check
+                if chainindex > chainlimit/2:
+                    try:
+                        self.nodes[0].sendtoaddress(chain_addrs[0], node0_balance, "", "", True)
+                        assert(False)
+                    except JSONRPCException as e:
+                        assert_equal(node0_balance, self.nodes[0].getbalance())
+                        if chainindex == chainlimit-2:
+                            # Last one is too long for each input as well
+                            assert("Insufficient" in e.error['message'])
+                        else:
+                            assert("too long" in e.error['message'])
+
+        # Or each individually
+        for i in range(2):
+            self.nodes[0].lockunspent(False, [{"txid":txids[(i+1)%2], "vout":0}])
+            amount = self.nodes[0].listunspent(0, 0, [chain_addrs[i]])[0]["amount"]
+            try:
+                txids[i] = self.nodes[0].sendtoaddress(chain_addrs[i], amount, "", "", True)
+                assert(False)
+            except JSONRPCException as e:
+                assert("Insufficient funds" in e.error['message'])
+            self.nodes[0].lockunspent(True, [{"txid":txids[(i+1)%2], "vout":0}])
+
+        assert_equal(node0_balance, self.nodes[0].getbalance("*", 0))
```

Comment:
Particularly I was interested in checking it errors out when, and how. So I build two long chains, start checking that sending will fail due to final checks when combining the chains, not "lack of coins", then once it gets long enough, both coins are already too long to include.

This test might be too obscure/specific, but it helped convince me that the code is doing what I wanted it to do. I should probably add a simple test like @sdaftuar 's where it shows it preferentially choosing shorter chained outputs first.

>please don't use getbalance("*")

Leftover, sorry. Will fix.